docs/DeveloperGuide.md

```diff
@@ -153,6 +153,137 @@ The following sequence diagram shows how the undo operation works:
 
 _{Explain here how the data archiving feature will be implemented}_
 
+### Add Ingredient feature
+
+#### Implementation
+This feature allows users to add ingredients into the fridge.
+
+Substitutability is used in Command and Parser:
+* `AddIngredientCommand` extends `Command`
+* `AddIngredientCommandParser` implements `Parser<AddIngredientCommand>`
+
+Given below is an example usage scenario and how the mechanism behaves at each step.
+
+![AddIngredientSequence](images/AddIngredientSequence.png)
+
+Step 1:
+User inputs the add ingredient command to add ingredients into the fridge.
+
+Step 2:
+After successful parsing of user input, the `AddIngredientCommand#execute(Model model)` method is called.
+
+Step 3:
+The ingredients that the user has input will be saved into Wishful Shrinking's fridge of ingredients.
+
+Step 4:
+After the successful adding of the ingredient, a `CommandResult` object is instantiated and returned to `LogicManager`.
+
+#### Design Considerations
+##### Aspect 1: Concern while adding a new feature
+* Workflow must be consistent with other adding commands e.g. add recipe and eat recipe for consumption.
+
+##### Aspect 2: How do we successfully parse the ingredients the user has added with the optional ingredient quantity 
+* **Alternative 1 (current choice):** Add a quantity field in the Ingredient class as well as a IngredientParser class to parse each ingredient added
+  * Pros: Easy to implement.
+  * Cons: The parser may confuse ingredients that have prefixes that Wishful Shrinking uses to identify fields in the names, eg "-" or ","
+
+### Search Recipe feature
+
+#### Implementation
+This feature allows users search for recipes in the recipe list based on the name, tags or ingredients.
+
+Substitutability is used in Command and Parser:
+* `SearchRecipeCommand` extends `Command`
+* `SearchRecipeCommandParser` implements `Parser<SearchRecipeCommand>`
+
+Given below is an example usage scenario and how the mechanism behaves at each step.
+
+![SearchRecipeSequence](images/SearchRecipeSequence.png)
+
+Step 1:
+User inputs the search recipe command to search for the recipes they want.
+
+Step 2:
+After successful parsing of user input, the `SearchRecipeCommand#execute(Model model)` method is called.
+
+Step 3:
+The list of recipes that fit the user's search will be returned to the user.
+
+Step 4:
+After the successful searching of the recipes, a `CommandResult` object is instantiated and returned to `LogicManager`.
```

Comment:
Is this the sequence diagram explanation?

Reply:
yes brief explanation, but will change if need